resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/ExecutorPodsSnapshotsStoreImpl.scala

```diff
@@ -57,10 +60,22 @@ import org.apache.spark.util.ThreadUtils
  * The subscriber notification callback is guaranteed to be called from a single thread at a time.
  */
 private[spark] class ExecutorPodsSnapshotsStoreImpl(
+    conf: SparkConf,
     subscribersExecutor: ScheduledExecutorService,
     clock: Clock = new SystemClock)
   extends ExecutorPodsSnapshotsStore with Logging {
 
+  private[spark] def this(
+      subscribersExecutor: ScheduledExecutorService) = {
+    this(new SparkConf, subscribersExecutor, new SystemClock)
+  }
+
+  private[spark] def this(
+      subscribersExecutor: ScheduledExecutorService,
+      clock: Clock) = {
+    this(new SparkConf, subscribersExecutor, clock)
+  }
```

Comment:
I can not merge these two constructers into one,
```
  private[spark] def this(
      subscribersExecutor: ScheduledExecutorService,
      clock: Clock = new SystemClock) = {
    this(new SparkConf, subscribersExecutor, clock)
  }
```
it fails compilation
```
[error] /Users/chengpan/Projects/apache-spark/resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/KubernetesClusterManager.scala:106:64: type mismatch;
[error]  found   : org.apache.spark.SparkConf
[error]  required: java.util.concurrent.ScheduledExecutorService
[error]     val snapshotsStore = new ExecutorPodsSnapshotsStoreImpl(sc.conf, subscribersExecutor)
[error]
``

Reply:
Oh, interesting.

Reply:
> I can not merge these two constructers into one,
> 
> ```
>   private[spark] def this(
>       subscribersExecutor: ScheduledExecutorService,
>       clock: Clock = new SystemClock) = {
>     this(new SparkConf, subscribersExecutor, clock)
>   }
> ```
> 
> it fails compilation
> 
> ```
> [error] /Users/chengpan/Projects/apache-spark/resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/KubernetesClusterManager.scala:106:64: type mismatch;
> [error]  found   : org.apache.spark.SparkConf
> [error]  required: java.util.concurrent.ScheduledExecutorService
> [error]     val snapshotsStore = new ExecutorPodsSnapshotsStoreImpl(sc.conf, subscribersExecutor)
> [error]
> ``
> ```

Need remove the default value of `SystemClock` from primary constructor if you want to merge them to one

Reply:
Then `clock` is required if user wants to pass `conf`, it's not friendly IMO.